### PR TITLE
Marshall/op e1s power seq

### PR DESCRIPTION
**Pull request by BonnieLo**
Summary:
- Support oem command to control e1s.
- Implement e1s power onoff sequence.

Test Plan:
-Build code: Pass

### CODE REVIEW

File: meta-facebook/op2-op/src/platform/plat_power_seq.c [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+	[4] = { .present = OPB_E1S_4_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_4_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_4_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_4_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_4_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_4_OE_N,
+		.pcie_reset = OPB_RST_E1S_4_PERST },
+};
+
+uint8_t get_e1s_present(uint8_t device_index)
```
Comment (garnermic) [inline meta-facebook/op2-op/src/platform/plat_power_seq.c:99]: Presence should be a bool and not a uint8_t as the drive is either present or not.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+		present = gpio_get(opb_e1s_power_control_gpio[device_index].present);
+		break;
+	default:
+		LOG_ERR("UNKNOWN CARD TYPE");
+		break;
+	}
+	return present;
+}
+
+uint8_t get_e1s_power_good(uint8_t device_index)
```
Comment (garnermic) [inline meta-facebook/op2-op/src/platform/plat_power_seq.c:118]: power good should be a bool as it is either good or not.  This should not be defined as a uint8_t.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+		}
+		break;
+	default:
+		LOG_ERR("Check mode 0x%x not supported!", check_mode);
+		ret = 1;
+		break;
+	}
+
+	if (ret == -1) { // Addsel if check power stage fail
+		//TBD
```
Comment (garnermic) [inline meta-facebook/op2-op/src/platform/plat_power_seq.c:185]: If this is an error condition, at a minimum it should log it as such.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+	uint8_t card_type = get_card_type();
+	if (card_type == CARD_TYPE_UNKNOWN) {
+		LOG_ERR("UNKNOWN CARD TYPE");
+		return false;
+	}
+
+	while (enable_power_on_handler == true) {
+		switch (control_stage) {
+		case E1S_POWER_ON_STAGE0:
+			//checking the device present status
```
Comment (garnermic) [inline meta-facebook/op2-op/src/platform/plat_power_seq.c:209]: Is this missing code here?  Shouldn't there be a presence check based on the comment?

**Review by garnermic — CHANGES_REQUESTED** [review]

File: meta-facebook/op2-op/src/platform/plat_power_seq.c (hunk 4 above)
Comment (MarshallZhan-wiwynn) [reply meta-facebook/op2-op/src/platform/plat_power_seq.c]: BIC will skip the first control stage, and check the present status in check stage.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c (hunk 3 above)
Comment (MarshallZhan-wiwynn) [reply meta-facebook/op2-op/src/platform/plat_power_seq.c]: Modify as recommendation.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c (hunk 2 above)
Comment (MarshallZhan-wiwynn) [reply meta-facebook/op2-op/src/platform/plat_power_seq.c]: Modify as recommendation.

File: meta-facebook/op2-op/src/platform/plat_power_seq.c (hunk 1 above)
Comment (MarshallZhan-wiwynn) [reply meta-facebook/op2-op/src/platform/plat_power_seq.c]: Modify as recommendation.

**Comment by MarshallZhan-wiwynn** [comment]
Fix merge conflict and clang-format

**Pull request closed without merging**